programs/drift/src/instructions/keeper.rs

```diff
@@ -2930,6 +2947,376 @@ pub fn handle_pause_spot_market_deposit_withdraw(
+    let constituent_token_account = &mut ctx.accounts.constituent_quote_token_account;
+    let mut lp_pool = ctx.accounts.lp_pool.load_mut()?;
+
+    let clock = Clock::get()?;
+
+    let expected_pda = &Pubkey::create_program_address(
+        &[
+            AMM_POSITIONS_CACHE.as_ref(),
+            amm_cache.fixed.bump.to_le_bytes().as_ref(),
+        ],
+        &crate::ID,
+    )
+    .map_err(|_| ErrorCode::InvalidPDA)?;
+    validate!(
+        expected_pda.eq(amm_cache_key),
+        ErrorCode::InvalidPDA,
+        "Amm cache PDA does not match expected PDA"
+    )?;
+
+    let remaining_accounts_iter = &mut ctx.remaining_accounts.iter().peekable();
+    let AccountMaps {
+        perp_market_map,
+        spot_market_map: _,
+        oracle_map: _,
+    } = load_maps(
+        remaining_accounts_iter,
+        &MarketSet::new(),
+        &MarketSet::new(),
+        slot,
+        None,
+    )?;
+
+    for (_, perp_market_loader) in perp_market_map.0.iter() {
+        let mut perp_market = perp_market_loader.load_mut()?;
+        let cached_info = amm_cache.get_mut(perp_market.market_index as u32);
+
+        if slot.saturating_sub(cached_info.oracle_slot) > SETTLE_AMM_ORACLE_MAX_DELAY {
+            // If the oracle slot is not up to date, skip this market
+            msg!(
+                "Skipping settling perp market {} to dlp because oracle slot is not up to date",
+                perp_market.market_index
+            );
+            continue;
+        }
+
+        let fee_pool_token_amount = get_token_amount(
+            perp_market.amm.fee_pool.scaled_balance,
+            &quote_market,
+            perp_market.amm.fee_pool.balance_type(),
+        )?;
+
+        let net_pnl_pool_token_amount = get_token_amount(
+            perp_market.pnl_pool.scaled_balance,
+            &quote_market,
+            perp_market.pnl_pool.balance_type(),
+        )?
+        .cast::<i128>()?
+        .safe_sub(calculate_net_user_pnl(
+            &perp_market.amm,
+            cached_info.oracle_price,
+        )?)?;
+
+        let amm_amount_available =
+            net_pnl_pool_token_amount.safe_add(fee_pool_token_amount.cast::<i128>()?)?;
+
+        if cached_info.last_net_pnl_pool_token_amount == 0
+            && cached_info.last_fee_pool_token_amount == 0
+        {
+            cached_info.last_fee_pool_token_amount = fee_pool_token_amount;
+            cached_info.last_net_pnl_pool_token_amount = net_pnl_pool_token_amount;
+            continue;
+        }
+
+        validate!(
+            perp_market.oracle_id() == cached_info.oracle_id()?,
+            ErrorCode::DefaultError,
+            "oracle id mismatch between amm cache and perp market"
+        )?;
+
+        // Actually transfer the pnl to the lp usdc constituent account
+        validate_market_within_price_band(&perp_market, state, cached_info.oracle_price)?;
+
+        if perp_market.is_operation_paused(PerpOperation::SettlePnl) {
```

Comment:
do we need a flag on one of the lp accounts to pause pnl settling at per market level?

Reply:
would have a flag to pause settling lp pnl without settling user pnl